.jenkins/jenkins.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash -f
+
+### Some environment variables available from Jenkins
+### Note: for a complete list see https://jenkins.ginko.ch/env-vars.html
+# slave              The name of the build slave (daint, kesch, ...).
+# BUILD_NUMBER       The current build number, such as "153".
+# BUILD_ID           The current build id, such as "2005-08-22_23-59-59" (YYYY-MM-DD_hh-mm-ss).
+# BUILD_DISPLAY_NAME The display name of the current build, something like "#153" by default.
+# NODE_NAME          Name of the slave if the build is on a slave, or "master" if run on master.
+# NODE_LABELS        Whitespace-separated list of labels that the node is assigned.
+# JENKINS_HOME       The absolute path of the data storage directory assigned on the master node.
+# JENKINS_URL        Full URL of Jenkins, like http://server:port/jenkins/
+# BUILD_URL          Full URL of this build, like http://server:port/jenkins/job/foo/15/
+# JOB_URL            Full URL of this job, like http://server:port/jenkins/job/foo/
+
+exitError()
+{
+    echo "ERROR $1: $3" 1>&2
+    echo "ERROR     LOCATION=$0" 1>&2
+    echo "ERROR     LINE=$2" 1>&2
+    exit $1
+}
+
+# echo basic setup
+echo "####### executing: $0 $* (PID=$$ HOST=$HOSTNAME TIME=`date '+%D %H:%M:%S'`)"
+
+# start timer
+T="$(date +%s)"
+
+# check sanity of environment
+test -n "$1" || exitError 1001 ${LINENO} "must pass an argument"
+test -n "${slave}" || exitError 1005 ${LINENO} "slave is not defined"
+shortslave=`echo ${slave} | sed 's/[0-9]*$//g'`
+echo "$1" | egrep '^test$' || exitError 1007 ${LINENO} "invalid argument (must be test)"
+
+# some global variables
+action="$1"
+optarg="$2"
+
+# check presence of env directory
+pushd `dirname $0` > /dev/null
+envloc=`/bin/pwd`
+popd > /dev/null
+
+# Download the env
+. ${envloc}/env.sh
+
+# setup module environment and default queue
+test -f ${envloc}/env/machineEnvironment.sh || exitError 1201 ${LINENO} "cannot find machineEnvironment.sh script"
+. ${envloc}/env/machineEnvironment.sh
+
+# check that host (define in machineEnvironment.sh) and slave are consistent
+echo ${host} | grep "${shortslave}" || exitError 1006 ${LINENO} "host does not contain slave"
+
+# act according to action specified
+case "${action}" in
+
+test )
+
+    # check if test script exists
+    script=".jenkins/test.sh"
```

Comment:
should we move directories here to ensure this works from anywhere and not only root?